src/protocolgame.cpp

```diff
@@ -3426,15 +3428,16 @@ void ProtocolGame::sendSessionEnd(SessionEndTypes_t reason)
 }
 
 ////////////// Add common messages
-void ProtocolGame::AddCreature(NetworkMessage& msg, const Creature* creature, bool known, uint32_t remove)
+void ProtocolGame::AddCreature(NetworkMessage& msg, std::shared_ptr<const Creature> creature, bool known,
+                               uint32_t remove)
 {
 	CreatureType_t creatureType = creature->getType();
-	const Player* otherPlayer = creature->getPlayer();
-	const Player* masterPlayer = nullptr;
+	auto otherPlayer = creature->getPlayer();
+	std::shared_ptr<const Player> masterPlayer = nullptr;
 	uint32_t masterId = 0;
 
 	if (creatureType == CREATURETYPE_MONSTER) {
-		const Creature* master = creature->getMaster();
+		auto master = creature->getMaster();
```

Comment:
Could this be `const auto`
```suggestion
		const auto master = creature->getMaster();
```